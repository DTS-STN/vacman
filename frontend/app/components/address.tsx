import type { ComponentProps, JSX } from 'react';

import { formatAddress } from '~/utils/string-utils';
import { cn } from '~/utils/tailwind-utils';

type AddressProps = ComponentProps<'address'> & {
  address: Parameters<typeof formatAddress>[0];
  format?: Parameters<typeof formatAddress>[1];
};

export function Address({ address, format, className, ...rest }: AddressProps): JSX.Element {
  return (
    <address className={cn('whitespace-pre-wrap not-italic', className)} data-testid="address-id" {...rest}>
      {formatAddress(address, format)}
    </address>
  );
}
