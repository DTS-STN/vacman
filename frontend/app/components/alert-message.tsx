import type { Ref } from 'react';

import type { IconDefinition } from '@fortawesome/free-solid-svg-icons';
import { faTriangleExclamation, faCircleExclamation, faCircleCheck } from '@fortawesome/free-solid-svg-icons';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';

interface AlertMessageProps {
  message: string;
  type: 'success' | 'info' | 'error';
  ref?: Ref<HTMLDivElement>;
}

const styles: Record<NonNullable<AlertMessageProps['type']>, string> = {
  success: `border-green-600 bg-green-100`,
  error: `border-red-800 bg-red-100`,
  info: `border-sky-700 bg-sky-100`,
};

const icons: Record<NonNullable<AlertMessageProps['type']>, IconDefinition> = {
  success: faCircleCheck,
  error: faTriangleExclamation,
  info: faCircleExclamation,
};

export function AlertMessage({ ref, message, type = 'info' }: AlertMessageProps) {
  return (
    <div ref={ref} className={`${styles[type]} flex w-full items-center border-l-4 p-2`}>
      <FontAwesomeIcon
        icon={icons[type]}
        className={`${type === 'success' ? 'text-green-600' : type === 'error' ? 'text-red-800' : 'text-sky-700'} mr-2`}
      />
      {message}
    </div>
  );
}
