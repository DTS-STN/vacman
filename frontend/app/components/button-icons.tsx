import type { FontAwesomeIconProps } from '@fortawesome/react-fontawesome';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';

import { cn } from '~/utils/tailwind-utils';

export function ButtonStartIcon({ className, ...restProps }: FontAwesomeIconProps) {
  return <FontAwesomeIcon fixedWidth className={cn('me-1', className)} {...restProps} />;
}

export function ButtonEndIcon({ className, ...restProps }: FontAwesomeIconProps) {
  return <FontAwesomeIcon fixedWidth className={cn('ms-1', className)} {...restProps} />;
}
