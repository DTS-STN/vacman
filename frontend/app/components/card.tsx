import type { ComponentProps, JSX } from 'react';

import type { FlipProp, IconProp } from '@fortawesome/fontawesome-svg-core';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { Slot } from '@radix-ui/react-slot';

import { cn } from '~/utils/tailwind-utils';

type CardProps = ComponentProps<'div'> & {
  /**
   * Use the asChild prop to change the default rendered element for the one passed
   * as a child, merging their props and behavior. Useful for rendering links.
   */
  asChild?: boolean;
};

/**
 * Card component renders a card with optional child rendering.
 * It supports various customization options.
 */
export function Card({ className, asChild, ...props }: CardProps): JSX.Element {
  const Component = asChild ? Slot : 'div';

  return (
    <Component
      className={cn(
        'group block rounded-xs border bg-white shadow-xs [&:is(a)]:hover:bg-gray-50 [&:is(a)]:hover:shadow-md',
        className,
      )}
      {...props}
    />
  );
}

/**
 * CardHeader component renders the header section of a card.
 */
export function CardHeader({ className, ...props }: ComponentProps<'div'>) {
  return <div className={cn('flex flex-col space-y-1.5 p-6', className)} {...props} />;
}

type CardTitleProps = ComponentProps<'h1' | 'h2' | 'h3' | 'h4' | 'h5' | 'h6'> & {
  /**
   * Use the asChild prop to change the default rendered element for the one passed as a child, merging their props and behavior.
   */
  asChild?: boolean;
};

/**
 * CardTitle component renders the title of a card.
 * It supports optional child rendering.
 */
export function CardTitle({ asChild, className, ...props }: CardTitleProps) {
  const Component = asChild ? Slot : 'h3';

  return (
    <Component
      className={cn(
        'leading-none font-semibold group-[&:is(a)]:text-slate-700 group-[&:is(a)]:underline group-[&:is(a):focus]:text-blue-700 group-[&:is(a):hover]:text-blue-700',
        className,
      )}
      {...props}
    />
  );
}

/**
 * CardDescription component renders the description of a card.
 */
export function CardDescription({ className, ...props }: ComponentProps<'div'>) {
  return <div className={cn('text-sm text-black/60', className)} {...props} />;
}

/**
 * CardContent component renders the content section of a card.
 */
export function CardContent({ className, ...props }: ComponentProps<'div'>) {
  return <div className={cn('p-6 pt-0', className)} {...props} />;
}

/**
 * CardFooter component renders the footer section of a card.
 */
export function CardFooter({ className, ...props }: ComponentProps<'div'>) {
  return <div className={cn('flex items-center p-6 pt-0', className)} {...props} />;
}

/**
 * CardImage component renders an image in a card.
 */
export function CardImage({ className, ...props }: ComponentProps<'img'>) {
  return (
    <img
      alt=""
      className={cn('relative aspect-video h-40 w-full object-cover sm:h-48', className)}
      loading="lazy"
      draggable="false"
      {...props}
    />
  );
}

/**
 * CardTag component renders a tag in a card.
 */
export function CardTag({ className, ...props }: ComponentProps<'span'>) {
  return <span className={cn('inline-block bg-cyan-700 px-2 py-1 text-xs text-white', className)} {...props} />;
}

type CardIconProps = OmitStrict<ComponentProps<'div'>, 'children'> & {
  /**
   * FontAwesome icon to be displayed.
   */
  icon: IconProp;
  /**
   * Additional class names for styling the icon.
   */
  iconClassName?: IconProp;
  /**
   * Flip property for the icon.
   */
  iconFlip?: FlipProp;
};

/**
 * CardIcon component renders an icon in a card.
 */
export function CardIcon({ className, icon, iconClassName, iconFlip, ...props }: CardIconProps) {
  return (
    <div className="flex size-10 shrink-0 items-center justify-center rounded-full bg-slate-700 sm:size-12" {...props}>
      <FontAwesomeIcon icon={icon} flip={iconFlip} className={cn('size-4 text-white sm:size-5', iconClassName)} />
    </div>
  );
}
