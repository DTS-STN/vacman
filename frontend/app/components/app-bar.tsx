import type { JSX } from 'react';

import { faChevronDown, faRightFromBracket, faUser } from '@fortawesome/free-solid-svg-icons';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { useTranslation } from 'react-i18next';

import { DropdownMenu, DropdownMenuContent, DropdownMenuLabel, DropdownMenuTrigger } from '~/components/dropdown-menu';
import { LanguageSwitcher } from '~/components/language-switcher';
import { Menu, MenuItem } from '~/components/menu';
import { cn } from '~/utils/tailwind-utils';

type AppBarProps = {
  children: React.ReactNode;
  name?: string;
  profileItems?: React.ReactNode;
};

export function AppBar({ children, name, profileItems }: AppBarProps): JSX.Element {
  const { t } = useTranslation(['gcweb']);

  return (
    <div className="bg-slate-700">
      <div className="align-center container mx-auto flex flex-wrap justify-between">
        <div className="align-center flex">
          <Menu>{children}</Menu>
        </div>
        <div className="flex items-center space-x-4 text-right">
          {name && <UserButton name={name}>{profileItems}</UserButton>}
          <LanguageSwitcher>{t('gcweb:language-switcher.alt-lang')}</LanguageSwitcher>
        </div>
      </div>
    </div>
  );
}

type UserButtonProps = {
  className?: string;
  children?: React.ReactNode;
  name?: string;
};

function UserButton({ className, children, name }: UserButtonProps): JSX.Element {
  const { t } = useTranslation(['gcweb']);

  return (
    <DropdownMenu>
      <DropdownMenuTrigger
        className={cn(
          'flex h-full flex-nowrap space-x-2 bg-slate-600 px-2 text-sm text-white hover:bg-slate-500 hover:underline focus:ring-2 focus:ring-black focus:ring-offset-2 focus:outline-hidden aria-expanded:bg-slate-800 aria-expanded:text-white sm:space-x-4 sm:px-4',
          className,
        )}
      >
        <div className="text-md my-auto flex flex-nowrap items-center space-x-2 py-2">
          <div className="flex h-8 w-8 items-center justify-center rounded-full bg-white">
            <FontAwesomeIcon icon={faUser} className="size-5 text-slate-700" />
          </div>
          <span id="menu-label" className="text-md hidden py-2 font-bold sm:block">
            {name}
          </span>
        </div>
        <FontAwesomeIcon icon={faChevronDown} className="my-auto size-4" />
      </DropdownMenuTrigger>
      <DropdownMenuContent align="end" className="w-64 bg-slate-700">
        <UserName name={name} />
        <MenuItem file="routes/protected/index.tsx">{t('gcweb:app.profile')}</MenuItem>
        {children}
        <MenuItem to="/auth/logout" className="text-md flex justify-between">
          {t('gcweb:app.logout')}
          <FontAwesomeIcon icon={faRightFromBracket} className="my-auto size-8" />
        </MenuItem>
      </DropdownMenuContent>
    </DropdownMenu>
  );
}

type UserNameProps = {
  name?: string;
};

function UserName({ name }: UserNameProps): JSX.Element {
  return (
    <>
      {name !== undefined && (
        <DropdownMenuLabel className="text-md flex items-center border-b-2 border-slate-600 px-3 py-2 text-gray-300 sm:hidden">
          <FontAwesomeIcon icon={faUser} className="mr-2 size-4" />
          {name}
        </DropdownMenuLabel>
      )}
    </>
  );
}
