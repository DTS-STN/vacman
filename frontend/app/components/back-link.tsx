import type { Params, Path } from 'react-router';

import { faAngleLeft } from '@fortawesome/free-solid-svg-icons';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { useTranslation } from 'react-i18next';

import { InlineLink } from '~/components/links';
import type { I18nRouteFile } from '~/i18n-routes';
import { cn } from '~/utils/tailwind-utils';

interface BackLinkProps {
  file: I18nRouteFile;
  params?: Params;
  className?: string;
  hash?: Path['hash'];
  search?: Path['search'];
  translationKey?: string;
}

export function BackLink({ className, file, params, hash, search, translationKey = 'app:profile.back' }: BackLinkProps) {
  const { t } = useTranslation();
  const linkText = t(translationKey);
  return (
    <InlineLink
      className={cn('inline-flex items-center', className)}
      file={file}
      params={params}
      hash={hash}
      search={search}
      aria-label={linkText}
    >
      <FontAwesomeIcon icon={faAngleLeft} />
      {linkText}
    </InlineLink>
  );
}
