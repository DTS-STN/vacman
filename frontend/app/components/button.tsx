import type { ComponentProps } from 'react';

import { ButtonEndIcon, ButtonStartIcon } from '~/components/button-icons';
import { cn } from '~/utils/tailwind-utils';

type ButtonEndIconProps = ComponentProps<typeof ButtonEndIcon>;
type ButtonStartIconProps = ComponentProps<typeof ButtonStartIcon>;

const sizes = {
  xs: 'px-3 py-2 text-xs',
  sm: 'px-3 py-2 text-sm',
  base: 'px-5 py-2.5 text-sm',
  lg: 'px-5 py-3 text-base',
  xl: 'px-6 py-3.5 text-base',
} as const;

// prettier-ignore
const variants = {
  alternative: 'border-gray-200 bg-white text-gray-900 hover:bg-gray-100 hover:text-blue-700 focus:bg-gray-100 focus:text-blue-700',
  default: 'border-gray-300 bg-gray-200 text-slate-700 hover:bg-neutral-300 focus:bg-neutral-300',
  dark: 'border-gray-800 bg-gray-800 text-white hover:bg-gray-900 focus:bg-gray-900',
  green: 'border-green-700 bg-green-700 text-white hover:bg-green-800 focus:bg-green-800',
  primary: 'border-slate-700 bg-slate-700 text-white hover:bg-sky-800 focus:bg-sky-800',
  red: 'border-red-700 bg-red-700 text-white hover:bg-red-800 focus:bg-red-800',
  link: 'border-0 bg-white text-gray-900 underline hover:bg-gray-100 hover:text-blue-700 focus:bg-gray-100 focus:text-blue-700',
  ghost: 'border-0 hover:bg-neutral-100 hover:text-neutral-900',
  outline: "border border-neutral-200 bg-white hover:bg-neutral-100 hover:text-neutral-900 disabled:text-neutral-500",
} as const;

export interface ButtonProps extends ComponentProps<'button'> {
  endIcon?: ButtonEndIconProps['icon'];
  endIconProps?: OmitStrict<ButtonEndIconProps, 'icon'>;
  pill?: boolean;
  size?: keyof typeof sizes;
  startIcon?: ButtonStartIconProps['icon'];
  startIconProps?: OmitStrict<ButtonStartIconProps, 'icon'>;
  variant?: keyof typeof variants;
}

/**
 * Tailwind CSS Buttons from Flowbite
 * @see https://flowbite.com/docs/components/buttons/
 */
export function Button({
  children,
  className,
  endIcon,
  endIconProps,
  pill,
  size = 'base',
  startIcon,
  startIconProps,
  variant = 'default',
  ...props
}: ButtonProps) {
  const buttonClassName = cn(
    'font-lato inline-flex items-center justify-center rounded border align-middle outline-offset-4',
    sizes[size],
    variants[variant],
    'disabled:pointer-events-none disabled:cursor-not-allowed disabled:opacity-70',
    pill && 'rounded-full',
    className,
  );
  return (
    <button className={buttonClassName} {...props}>
      {startIcon && <ButtonStartIcon {...(startIconProps ?? {})} icon={startIcon} />}
      {children}
      {endIcon && <ButtonEndIcon {...(endIconProps ?? {})} icon={endIcon} />}
    </button>
  );
}
