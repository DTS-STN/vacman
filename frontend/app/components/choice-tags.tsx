import { faXmark } from '@fortawesome/free-solid-svg-icons';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { useTranslation } from 'react-i18next';

import { Button } from '~/components/button';

export interface ChoiceTag {
  ariaLabel?: string;
  label: string;
  name: string;
  value: string;
  group?: string;
}

export type ClearAllEventHandler = () => void;
export type DeleteEventHandler = (name: string, label: string, value: string, group?: string) => void;

export interface ChoiceTagsProps {
  choiceTags: ChoiceTag[];
  onClearAll?: ClearAllEventHandler;
  onDelete?: DeleteEventHandler;
}

export function ChoiceTags({ choiceTags, onClearAll, onDelete }: ChoiceTagsProps) {
  const { t } = useTranslation(['gcweb']);

  const handleOnDelete = (_event: unknown, choice: ChoiceTag) => {
    if (onDelete) onDelete(choice.name, choice.label, choice.value, choice.group);
  };
  const handleOnClearAll = () => {
    if (onClearAll) onClearAll();
  };
  return (
    <div id={`selected-${choiceTags[0]?.name}-wrapper`} className="flex flex-wrap gap-3">
      {choiceTags.map((choiceTag) => (
        <div
          key={choiceTag.value}
          className="inline-flex items-center justify-center rounded-sm border-2 border-gray-900 bg-blue-100 px-2 py-1 align-middle break-all text-gray-900"
        >
          <span>{choiceTag.group ? `${choiceTag.group} - ${choiceTag.label}` : choiceTag.label}</span>
          <button
            aria-label={t('gcweb:choice-tag.choice-tag-added-aria-label', {
              item: choiceTag.name,
              choice: choiceTag.group ? `${choiceTag.group} - ${choiceTag.label}` : choiceTag.label,
            })}
            onClick={(e) => handleOnDelete(e, choiceTag)}
            type="button"
          >
            <FontAwesomeIcon icon={faXmark} className="ml-1" />
          </button>
        </div>
      ))}
      {choiceTags.length > 1 && (
        <Button variant="primary" id="clear-all-button" onClick={() => handleOnClearAll()}>
          {t('gcweb:choice-tag.clear-all')}
        </Button>
      )}
    </div>
  );
}
