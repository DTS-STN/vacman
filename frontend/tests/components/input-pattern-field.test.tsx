import { render } from '@testing-library/react';
import { describe, expect, it } from 'vitest';

import { InputPatternField } from '~/components/input-pattern-field';

describe('InputPatternField', () => {
  const testFormat = '### ### ###';

  it.each([
    ['800000002', '800 000 002'],
    ['800 000 002', '800 000 002'],
    ['800-000-002', '800 000 002'],
    ['800 000-002', '800 000 002'],
    ['800000 002', '800 000 002'],
    ['800000-002', '800 000 002'],
  ])('should render %s -> %s', (defaultValue, expected) => {
    const { container } = render(
      <InputPatternField id="test-id" name="test" label="label test" defaultValue={defaultValue} format={testFormat} />,
    );
    expect(container).toMatchSnapshot('expected html');
  });

  it('should render with help message', () => {
    const { container } = render(
      <InputPatternField
        id="test-id"
        name="test"
        label="label test"
        format={testFormat}
        defaultValue="800000002"
        helpMessageSecondary="help message"
      />,
    );
    expect(container).toMatchSnapshot('expected html');
  });

  it('should render with required', () => {
    const { container } = render(
      <InputPatternField id="test-id" name="test" label="label test" format={testFormat} defaultValue="800000002" required />,
    );
    expect(container).toMatchSnapshot('expected html');
  });

  it('should render with error message', () => {
    const { container } = render(
      <InputPatternField
        id="test-id"
        name="test"
        label="label test"
        format={testFormat}
        defaultValue="800000002"
        errorMessage="error message"
      />,
    );
    expect(container).toMatchSnapshot('expected html');
  });
});
