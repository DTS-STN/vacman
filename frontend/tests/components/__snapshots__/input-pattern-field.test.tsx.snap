// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`InputPatternField > should render 800 000 002 -> 800 000 002 > expected html 1`] = `
<div>
  <div
    data-testid="input-pattern-field-test-id"
    id="input-pattern-field-test-id"
  >
    <label
      class="block mb-2"
      for="test-id"
      id="input-pattern-field-test-id-label"
    >
      <span
        class="font-semibold"
      >
        label test
      </span>
    </label>
    <input
      aria-invalid="false"
      aria-labelledby="input-pattern-field-test-id-label"
      class="block rounded-lg border-gray-500 focus:border-blue-500 focus:outline-none focus:ring focus:ring-blue-500 disabled:bg-gray-100 disabled:pointer-events-none disabled:cursor-not-allowed disabled:opacity-70 read-only:bg-gray-100 read-only:pointer-events-none read-only:cursor-not-allowed read-only:opacity-70"
      data-testid="input-pattern-field"
      id="test-id"
      inputmode="numeric"
      name="test"
      type="text"
      value="800 000 002"
    />
  </div>
</div>
`;

exports[`InputPatternField > should render 800 000-002 -> 800 000 002 > expected html 1`] = `
<div>
  <div
    data-testid="input-pattern-field-test-id"
    id="input-pattern-field-test-id"
  >
    <label
      class="block mb-2"
      for="test-id"
      id="input-pattern-field-test-id-label"
    >
      <span
        class="font-semibold"
      >
        label test
      </span>
    </label>
    <input
      aria-invalid="false"
      aria-labelledby="input-pattern-field-test-id-label"
      class="block rounded-lg border-gray-500 focus:border-blue-500 focus:outline-none focus:ring focus:ring-blue-500 disabled:bg-gray-100 disabled:pointer-events-none disabled:cursor-not-allowed disabled:opacity-70 read-only:bg-gray-100 read-only:pointer-events-none read-only:cursor-not-allowed read-only:opacity-70"
      data-testid="input-pattern-field"
      id="test-id"
      inputmode="numeric"
      name="test"
      type="text"
      value="800 000 002"
    />
  </div>
</div>
`;

exports[`InputPatternField > should render 800-000-002 -> 800 000 002 > expected html 1`] = `
<div>
  <div
    data-testid="input-pattern-field-test-id"
    id="input-pattern-field-test-id"
  >
    <label
      class="block mb-2"
      for="test-id"
      id="input-pattern-field-test-id-label"
    >
      <span
        class="font-semibold"
      >
        label test
      </span>
    </label>
    <input
      aria-invalid="false"
      aria-labelledby="input-pattern-field-test-id-label"
      class="block rounded-lg border-gray-500 focus:border-blue-500 focus:outline-none focus:ring focus:ring-blue-500 disabled:bg-gray-100 disabled:pointer-events-none disabled:cursor-not-allowed disabled:opacity-70 read-only:bg-gray-100 read-only:pointer-events-none read-only:cursor-not-allowed read-only:opacity-70"
      data-testid="input-pattern-field"
      id="test-id"
      inputmode="numeric"
      name="test"
      type="text"
      value="800 000 002"
    />
  </div>
</div>
`;

exports[`InputPatternField > should render 800000 002 -> 800 000 002 > expected html 1`] = `
<div>
  <div
    data-testid="input-pattern-field-test-id"
    id="input-pattern-field-test-id"
  >
    <label
      class="block mb-2"
      for="test-id"
      id="input-pattern-field-test-id-label"
    >
      <span
        class="font-semibold"
      >
        label test
      </span>
    </label>
    <input
      aria-invalid="false"
      aria-labelledby="input-pattern-field-test-id-label"
      class="block rounded-lg border-gray-500 focus:border-blue-500 focus:outline-none focus:ring focus:ring-blue-500 disabled:bg-gray-100 disabled:pointer-events-none disabled:cursor-not-allowed disabled:opacity-70 read-only:bg-gray-100 read-only:pointer-events-none read-only:cursor-not-allowed read-only:opacity-70"
      data-testid="input-pattern-field"
      id="test-id"
      inputmode="numeric"
      name="test"
      type="text"
      value="800 000 002"
    />
  </div>
</div>
`;

exports[`InputPatternField > should render 800000-002 -> 800 000 002 > expected html 1`] = `
<div>
  <div
    data-testid="input-pattern-field-test-id"
    id="input-pattern-field-test-id"
  >
    <label
      class="block mb-2"
      for="test-id"
      id="input-pattern-field-test-id-label"
    >
      <span
        class="font-semibold"
      >
        label test
      </span>
    </label>
    <input
      aria-invalid="false"
      aria-labelledby="input-pattern-field-test-id-label"
      class="block rounded-lg border-gray-500 focus:border-blue-500 focus:outline-none focus:ring focus:ring-blue-500 disabled:bg-gray-100 disabled:pointer-events-none disabled:cursor-not-allowed disabled:opacity-70 read-only:bg-gray-100 read-only:pointer-events-none read-only:cursor-not-allowed read-only:opacity-70"
      data-testid="input-pattern-field"
      id="test-id"
      inputmode="numeric"
      name="test"
      type="text"
      value="800 000 002"
    />
  </div>
</div>
`;

exports[`InputPatternField > should render 800000002 -> 800 000 002 > expected html 1`] = `
<div>
  <div
    data-testid="input-pattern-field-test-id"
    id="input-pattern-field-test-id"
  >
    <label
      class="block mb-2"
      for="test-id"
      id="input-pattern-field-test-id-label"
    >
      <span
        class="font-semibold"
      >
        label test
      </span>
    </label>
    <input
      aria-invalid="false"
      aria-labelledby="input-pattern-field-test-id-label"
      class="block rounded-lg border-gray-500 focus:border-blue-500 focus:outline-none focus:ring focus:ring-blue-500 disabled:bg-gray-100 disabled:pointer-events-none disabled:cursor-not-allowed disabled:opacity-70 read-only:bg-gray-100 read-only:pointer-events-none read-only:cursor-not-allowed read-only:opacity-70"
      data-testid="input-pattern-field"
      id="test-id"
      inputmode="numeric"
      name="test"
      type="text"
      value="800 000 002"
    />
  </div>
</div>
`;

exports[`InputPatternField > should render with error message > expected html 1`] = `
<div>
  <div
    data-testid="input-pattern-field-test-id"
    id="input-pattern-field-test-id"
  >
    <label
      class="block mb-2"
      for="test-id"
      id="input-pattern-field-test-id-label"
    >
      <span
        class="font-semibold"
      >
        label test
      </span>
    </label>
    <p
      class="mb-2"
    >
      <span
        class="inline-block max-w-prose border-l-2 border-red-600 bg-red-50 px-3 py-1"
        data-testid="input-error-test-id"
        id="input-pattern-field-test-id-error"
        role="alert"
      >
        error message
      </span>
    </p>
    <input
      aria-errormessage="input-pattern-field-test-id-error"
      aria-invalid="true"
      aria-labelledby="input-pattern-field-test-id-label"
      class="block rounded-lg focus:outline-none focus:ring disabled:bg-gray-100 disabled:pointer-events-none disabled:cursor-not-allowed disabled:opacity-70 read-only:bg-gray-100 read-only:pointer-events-none read-only:cursor-not-allowed read-only:opacity-70 border-red-500 focus:border-red-500 focus:ring-red-500"
      data-testid="input-pattern-field"
      id="test-id"
      inputmode="numeric"
      name="test"
      type="text"
      value="800 000 002"
    />
  </div>
</div>
`;

exports[`InputPatternField > should render with help message > expected html 1`] = `
<div>
  <div
    data-testid="input-pattern-field-test-id"
    id="input-pattern-field-test-id"
  >
    <label
      class="block mb-2"
      for="test-id"
      id="input-pattern-field-test-id-label"
    >
      <span
        class="font-semibold"
      >
        label test
      </span>
    </label>
    <input
      aria-describedby="input-pattern-field-test-id-help-secondary"
      aria-invalid="false"
      aria-labelledby="input-pattern-field-test-id-label"
      class="block rounded-lg border-gray-500 focus:border-blue-500 focus:outline-none focus:ring focus:ring-blue-500 disabled:bg-gray-100 disabled:pointer-events-none disabled:cursor-not-allowed disabled:opacity-70 read-only:bg-gray-100 read-only:pointer-events-none read-only:cursor-not-allowed read-only:opacity-70"
      data-testid="input-pattern-field"
      id="test-id"
      inputmode="numeric"
      name="test"
      type="text"
      value="800 000 002"
    />
    <div
      class="block max-w-prose text-gray-500 mt-2"
      data-testid="input-pattern-field-help-secondary"
      id="input-pattern-field-test-id-help-secondary"
    >
      help message
    </div>
  </div>
</div>
`;

exports[`InputPatternField > should render with required > expected html 1`] = `
<div>
  <div
    data-testid="input-pattern-field-test-id"
    id="input-pattern-field-test-id"
  >
    <label
      class="block mb-2"
      for="test-id"
      id="input-pattern-field-test-id-label"
    >
      <span
        class="font-semibold"
      >
        label test
      </span>
       
      <span
        aria-hidden="true"
      >
        (
        input.required
        )
      </span>
    </label>
    <input
      aria-invalid="false"
      aria-labelledby="input-pattern-field-test-id-label"
      aria-required="true"
      class="block rounded-lg border-gray-500 focus:border-blue-500 focus:outline-none focus:ring focus:ring-blue-500 disabled:bg-gray-100 disabled:pointer-events-none disabled:cursor-not-allowed disabled:opacity-70 read-only:bg-gray-100 read-only:pointer-events-none read-only:cursor-not-allowed read-only:opacity-70"
      data-testid="input-pattern-field"
      id="test-id"
      inputmode="numeric"
      name="test"
      required=""
      type="text"
      value="800 000 002"
    />
  </div>
</div>
`;
